### PR TITLE
[ENHANCEMENT] Update dependencies and improve docs

### LINKED ISSUE #16: Needs Dependency Updates
The Go version and the dependencies of the perses-operator needs to be updated. For example, [perses dependency](https://github.com/perses/perses-operator/blob/b4c59f10020fd77d2eab601e333970b8ea208661/go.mod#L9) is still referring to the v0.44.0 version. Lot of schema changes have been made in the perses project since then. 
https://github.com/cloudoperators/greenhouse-extensions/issues/419

### PR DESCRIPTION
**Pull request by ibakshay**
## Changes
- Upgraded to Go 1.23
- Updated all project dependencies
- Enhanced README
- Refreshed config/samples manifests
- Updated CRDs in config/crd/bases to reflect new schema
- Added optional `perses-server-url` flag for localhost testing:
  ```go
  flag.StringVar(&persesServerURL, "perses-server-url", "", "The Perses backend server URL")
  ```

fixes #16 https://github.com/cloudoperators/greenhouse-extensions/issues/419

### CODE REVIEW

**Comment by jgbernalp** [comment]
@ibakshay thank you for your contribution!. Can you please take a look at the tests, they might need an update.

**Comment by ibakshay** [comment]
> @ibakshay thank you for your contribution!. Can you please take a look at the tests, they might need an update.

My pleasure. Thanks a lot for your review. The test is failing due to this [serialization issue](https://github.com/perses/perses-operator/issues/20). I have already created a PR in [prometheus/common](https://github.com/prometheus/common/pull/708) repository for the fix. The test should be successfully after the PR in prometheus/common gets merged.

File: config/crd/bases/perses.dev_perses.yaml [hunk 1]
```diff
@@ -291,9 +291,6 @@ spec:
                           disable_custom:
                             type: boolean
                           options:
-                            items:
-                              format: duration
-                              type: string
```
Comment (ibakshay) [inline config/crd/bases/perses.dev_perses.yaml]: @jgbernalp, As you can see, `make manifests` removes this piece of code. 
 [this is the annotations we have for this field](https://github.com/perses/perses/blob/3d34cf1b8849a8bf648e0d455d59b8bb32362c31/pkg/model/api/config/frontend.go#L40-L43)

File: api/v1alpha1/perses_dashboard.go [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
 )
 
 type Dashboard struct {
 	persesv1.DashboardSpec `json:",inline"`
 }
 
+// DeepCopyInto is a manually implemented deep copy function and this is required because:
+// 1. The embedded persesv1.DashboardSpec from the Perses project doesn't implement DeepCopyInto
+// 2. controller-gen can't automatically generate DeepCopy methods for types it doesn't own
 func (in *Dashboard) DeepCopyInto(out *Dashboard) {
```
[the remark this answers is not in the document]
Comment (ibakshay) [reply api/v1alpha1/perses_dashboard.go:16]: Thanks a lot for the pointer, Augustin. I have manually marshalled and unmarshalled as I was getting issue during deepcopying using `go-deepcopy` library. The github.com/brunoga/deep works great and I have fixed in this commit https://github.com/perses/perses-operator/pull/21/commits/4e2fa72dc4789295f43ab76c65d6adddd4ee3108 :)

File: internal/perses/common/perses_client_factory.go [hunk 3]
```diff
@@ -41,8 +54,13 @@ func NewWithURL(url string) PersesClientFactory {
...
 func (f *PersesClientFactoryWithURL) CreateClient(config persesv1alpha1.Perses) (v1.ClientInterface, error) {
-	restClient, err := perseshttp.NewFromConfig(perseshttp.RestConfigClient{
-		URL: f.url,
+	urStr := f.url
+	parsedURL, err := common.ParseURL(urStr)
+	if err != nil {
+		return nil, err
+	}
+	restClient, err := clientConfig.NewRESTClient(clientConfig.RestConfigClient{
+		URL: &common.URL{URL: parsedURL.URL},
```
[the remark this answers is not in the document]
Comment (ibakshay) [reply internal/perses/common/perses_client_factory.go]: fixed in https://github.com/perses/perses-operator/pull/21/commits/4e2fa72dc4789295f43ab76c65d6adddd4ee3108 :)

File: internal/perses/common/perses_client_factory.go [hunk 4]
```diff
@@ -19,10 +21,21 @@ func NewWithConfig() PersesClientFactory {
...
+	} else {
+		urlStr = fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", perses.Name, perses.Namespace, perses.Spec.ContainerPort)
+	}
+	parsedURL, err := common.ParseURL(urlStr)
+	if err != nil {
+		return nil, err
+	}
 
+	restClient, err := clientConfig.NewRESTClient(clientConfig.RestConfigClient{
+		URL: &common.URL{URL: parsedURL.URL},
```
Comment (Nexucis) [inline internal/perses/common/perses_client_factory.go:37]:
```suggestion
URL: parsedURL,
```

Comment (ibakshay) [reply internal/perses/common/perses_client_factory.go]: fixed in https://github.com/perses/perses-operator/pull/21/commits/3849b48dbf6df3e9a4f1392cdb7065acc248aa43 :)

File: internal/perses/common/perses_client_factory.go [hunk 5]
```diff
@@ -41,8 +54,13 @@ func NewWithURL(url string) PersesClientFactory {
 }
 
 func (f *PersesClientFactoryWithURL) CreateClient(config persesv1alpha1.Perses) (v1.ClientInterface, error) {
```
Comment (Nexucis) [inline internal/perses/common/perses_client_factory.go:56]: Since this method doesn't use the config I guess we can remove it

Comment (ibakshay) [reply internal/perses/common/perses_client_factory.go]: Sure! Removed in 9847bb16fe454de599ab37c596cfe4d4ede6ee30 commit :)

**Review by Nexucis — APPROVED** [review]

**Pull request merged**